Userland/Utilities/allocate.cpp

```diff
@@ -1,81 +1,83 @@
 /*
  * Copyright (c) 2018-2020, Andreas Kling <kling@serenityos.org>
+ * Copyright (c) 2023, Liav A. <liavalb@hotmail.co.il>
  *
  * SPDX-License-Identifier: BSD-2-Clause
  */
 
 #include <AK/Format.h>
 #include <AK/Optional.h>
+#include <LibCore/ArgsParser.h>
 #include <LibCore/ElapsedTimer.h>
 #include <LibMain/Main.h>
 #include <unistd.h>
 
-static void usage()
-{
-    warnln("usage: allocate [number [unit (B/KiB/MiB)]]");
-    exit(1);
-}
-
 enum class Unit {
     Bytes,
     KiB,
     MiB,
+    GiB,
 };
 
 ErrorOr<int> serenity_main(Main::Arguments arguments)
 {
-    int count = 50;
+    unsigned iterations_count = 10;
+    unsigned allocation_size = 100;
     auto unit = Unit::MiB;
-
-    if (arguments.argc >= 2) {
-        auto number = arguments.strings[1].to_uint();
-        if (!number.has_value()) {
-            usage();
-        }
-        count = number.value();
-    }
-
-    if (arguments.argc >= 3) {
-        if (arguments.strings[2] == "B")
-            unit = Unit::Bytes;
-        else if (arguments.strings[2] == "KiB")
-            unit = Unit::KiB;
-        else if (arguments.strings[2] == "MiB")
-            unit = Unit::MiB;
-        else
-            usage();
+    StringView chosen_unit {};
+
+    Core::ArgsParser args_parser;
+    args_parser.add_option(chosen_unit, "Storage Unit", "unit", 'u', "Storage Unit (Base 2)");
+    args_parser.add_option(iterations_count, "Sleep Iterations", "iterations", 'n', "Iterations count of calling sleep");
+    args_parser.add_option(allocation_size, "Allocation Size", "allocation-size", 'c', "Allocation size (in Bytes, KiB, MiB or GiB)");
+    args_parser.parse(arguments);
+
+    if (chosen_unit == "B"sv) {
+        unit = Unit::Bytes;
+    } else if (chosen_unit == "KiB"sv) {
+        unit = Unit::KiB;
+    } else if (chosen_unit == "MiB") {
+        unit = Unit::MiB;
+    } else if (chosen_unit == "GiB") {
+        unit = Unit::GiB;
+    } else {
```

Comment:
This will return an error if no `-u` argument is given. Would it be better to only return an error if `chosen_unit` is empty?